packages/data-store-typeorm/src/entities/document/create-document.ts

```diff
@@ -19,64 +26,66 @@ export async function createDocument({
   json: any;
   options?: any;
 }): Promise<WalletDocument> {
-  logger.debug(`Creating document with id ${json.id}...`);
-  if (json.id) {
-    const existingDocument = await getDocumentById({
-      dataStore,
-      id: json.id,
-    });
+  return writeMutex.runExclusive(async () => {
```

Comment:
we're sure this library gets translated properly when running on a device?

Reply:
does the mutex get released automatically if there's an error?

Reply:
yes, I did few testing using iOS and android devices. didn't notice any issues on that